internal/signprovider/ecdsa/sepc256k1/provider.go

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright 2024 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package sepc256k1
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"encoding/hex"
+	"github.com/dustinxie/ecc"
+	"github.com/project-alvarium/alvarium-sdk-go/pkg/config"
+	"math/big"
+	"os"
+)
+
+// provider is a receiver that encapsulates required dependencies.
+type provider struct{}
+
+// New is a factory function that returns an initialized provider.
+func New() *provider {
+	return &provider{}
+}
+
+func (p *provider) Sign(key config.KeyInfo, content []byte) (string, error) {
+	prvKeyBytes, err := os.ReadFile(key.Path)
+	if err != nil {
+		return "", err
+	}
+	privateKeyBytes, err := hex.DecodeString(string(prvKeyBytes))
```

Comment:
Are we expecting the private/public key to be provided in hex string format? the [sepc256k1 keys provided in this PR ](https://github.com/project-alvarium/alvarium-sdk-go/pull/64/files#diff-842180bb69544ff1f94ee31f626320154e3e76e3d93108b93233b367a6d92383) are in `.pem` format.

Reply:
I pulled the decode logic out of here to match the other provider implementation. This is probably something can be handled on the consumer side.